### ISSUE TITLE
Arcus.Security has a dependency on Microsoft.AspNetCore.App

### ISSUE THREAD

**Issue by fgheysels** [arcus-azure/arcus.security#189]
**Describe the bug**
The `Arcus.Security.Core` and `Arcus.Security.Providers.AzureKeyVault` projects both have a dependency on `Microsoft.AspNetCore.App` which is in my humble opinion, and as far as I know, not necessary.    (I presume other projects inside the solution have the same dependency, but I haven't checked them all).

This creates an unnecessary dependency which is not a problem or does not become apparent as long as you use the `Arcus.Security` library in ASP.NET core projects.
I've created a console application however which runs as a container and consumes a message bus.  This project uses `Arcus.Security` to retrieve secrets using `ISecretProvider`.   As this container doesn't require any ASP.NET core features, the container is based on the `mcr.microsoft.com/dotnet/core/runtime:3.1-alpine`, but since `Arcus.Security` has a dependency on ASP.NET Core, my container fails to start:

```
It was not possible to find any compatible framework version
The framework 'Microsoft.AspNetCore.App', version '3.1.0' was not found.
  - No frameworks were found.
```

I think it might be sufficient to just remove the dependency on AspNetCore in the csproj files:

```csproj
<ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp3.1'">
    <FrameworkReference Include="Microsoft.AspNetCore.App" />
</ItemGroup>
```` 

I think the above itemgroups can be removed.

**Comment by stijnmoreels** [comment]
Hmm, I thought that it was a previous concern that the double target framework on these projects would require us to use that framework reference. We have project templates that works fine without the ASP.NET Core reference [here](https://github.com/arcus-azure/arcus.templates/blob/master/src/Arcus.Templates.ServiceBus.Topic/Dockerfile). I should ask @tomkerkhove for more info on this 🤔 .

**Comment by tomkerkhove** [comment]
This is only required for .NET Core >3.x so we can remove it here.

The NuGet versioning scheme implies that as well: https://www.nuget.org/packages/Microsoft.AspNetCore.App/

**Comment by fgheysels** [comment]
> Hmm, I thought that it was a previous concern that the double target framework on these projects would require us to use that framework reference. We have project templates that works fine without the ASP.NET Core reference [here](https://github.com/arcus-azure/arcus.templates/blob/master/src/Arcus.Templates.ServiceBus.Topic/Dockerfile). I should ask @tomkerkhove for more info on this 🤔 .

Since we're not dependent upon asp.net core features in this project, it is not an issue here.   In Arcus.WebApi however, we do have that dependency.

**Comment by fgheysels** [comment]
Moreover, now that I think about it; the Arcus.Security project targets both netstandard2 and .netcore3.1, but I think that it would be sufficent for Arcus.Security to only target netstandard2 ?

**Comment by fgheysels** [comment]
> This is only required for .NET Core >3.x so we can remove it here.
> 
> The NuGet versioning scheme implies that as well: https://www.nuget.org/packages/Microsoft.AspNetCore.App/

But even if you have a .NET Core 3 app, you do not necessarily need the aspnet core packages ?

**Comment by tomkerkhove** [comment]
> > This is only required for .NET Core >3.x so we can remove it here.
> > The NuGet versioning scheme implies that as well: [nuget.org/packages/Microsoft.AspNetCore.App](https://www.nuget.org/packages/Microsoft.AspNetCore.App/)
> 
> But even if you have a .NET Core 3 app, you do not necessarily need the aspnet core packages ?

No not perse!

> Moreover, now that I think about it; the Arcus.Security project targets both netstandard2 and .netcore3.1, but I think that it would be sufficient for Arcus.Security to only target netstandard2?

Unfortunately not because .NET Core 3.x is netstandard2.1 (so far for semver) - https://docs.microsoft.com/en-us/dotnet/standard/net-standard#net-implementation-support

**Comment by tomkerkhove** [comment]
Closed via #190

**Comment by fgheysels** [comment]
> > > This is only required for .NET Core >3.x so we can remove it here.
> > > The NuGet versioning scheme implies that as well: [nuget.org/packages/Microsoft.AspNetCore.App](https://www.nuget.org/packages/Microsoft.AspNetCore.App/)
> > 
> > 
> > But even if you have a .NET Core 3 app, you do not necessarily need the aspnet core packages ?
> 
> No not perse!
> 
> > Moreover, now that I think about it; the Arcus.Security project targets both netstandard2 and .netcore3.1, but I think that it would be sufficient for Arcus.Security to only target netstandard2?
> 
> Unfortunately not because .NET Core 3.x is netstandard2.1 (so far for semver) - https://docs.microsoft.com/en-us/dotnet/standard/net-standard#net-implementation-support

I thought netstandard2 is compatible with .NET Core 3.x, as in: you can use netstandard2 libs in .NET Core 3.x apps.
